protocol/strict-derivation.md

```diff
@@ -0,0 +1,355 @@
+submitted. This should avoid out-of-order or gapped batches. It might require to implement some form
+of persistence in the transaction manager, since we cannot reliably recover all globally pending
+batcher transactions in the L1 network.
+
+Furthermore, the batcher will need to be made aware of the Steady Batch Derivation rules, namely that
+invalid batches will be derived as deposit-only blocks. So in case of a reorg, the batcher should
+e.g. wait on the sequencer it is connected to until it has derived all blocks from L1 in order to
+only start batching new blocks on top of the possibly derived deposit-only chain segment.
+
+And finally, there must be some detection of unexpected deposit-only chain derivation, e.g. by
+querying the sync-status repeatedly and matching it against the expected safe chain. As described
+above, in case of any discrepancy, the batcher should then stop batching and wait for the sequencer
+to fully derive up until the latest L1 batcher transactions, and only then continue batching.
+
+## Sync Start Simplifications
+
+Strict Batch Ordering leads to a simpler sync start algorithm. It may be sufficient to walk back
+the L1 chain until finding a batcher transaction that starts a new channel, i.e., containing a first
+frame. The strict ordering rules, as detailed above, would in such a case drop any pre-existing
+staging channel, and the batch queue should always be empty right after the pipeline origin
+progresses.
+
+# Steady Batch Derivation
+
+Steady Batch Derivation changes how invalid batches and payload attributes at different stages in
+the derivation pipeline are treated.
+- Batch queue: if a batch is found to be invalid at the batch queue stage (`BatchDrop`), it is immediately replaced by an empty
+  batch, leading to deposit-only (first batch of an epoch) or empty payload attributes.
+- Engine queue: if the payload attributes are found by the EL client to be
+  [`INVALID`](https://github.com/ethereum/go-ethereum/blob/7fd7c1f7dd9ba8d90399df2f080e4101ae37a255/beacon/engine/errors.go#L63)
+  they are replaced by deposit-only/empty payload attributes and reinserted into the EL.
+
+## L1 origin selection
+
+There's an open design space about how to select the L1 origin when deriving one or more
+deposit-only blocks. I see 3 sensible options:
+
+- Option 1: Eagerly advance the L1 origin. This solves for missing L1 slots and will implicitly and
+automatically maintain a good L1/L2 block ratio.
+  - This guaranteed fast deposit inclusion for users and is similar to how a live sequencer
+  operates.
+  - Option 1.a: is a variation of this, where an in-protocol validation depth is maintained, so that
+  there's a constant sequencer drift, e.g. of 4 L1 blocks. This is even more similar to how a live
+  sequencer operates. This has the benefit that there's minimal disruption to the way that L1
+  origins change when going in and out of auto-derivation.
```

Comment:
Option 1 feels better than 1.a currently, we will see during implementation, but go with Option 1 for now.